armadillo/src/main/java/org/molgenis/armadillo/controller/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,51 @@
+package org.molgenis.armadillo.controller;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.CONFLICT;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+import org.molgenis.armadillo.exceptions.DuplicateObjectException;
+import org.molgenis.armadillo.exceptions.DuplicateProjectException;
+import org.molgenis.armadillo.exceptions.FileProcessingException;
+import org.molgenis.armadillo.exceptions.StorageException;
+import org.molgenis.armadillo.exceptions.UnknownObjectException;
+import org.molgenis.armadillo.exceptions.UnknownProjectException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
```

Comment:
Ah, that is interesting, a form of centralized error handling?

Reply:
Yes this is the "preferred" way of exception handling in controllers because it centralises the exceptions and gives you a lot of control over the responses.

Reply:
Btw, the best way to return errors to the client would be to wrap them together with some other values, for example:

```json
{
  "error": 
  {
    "code": "404",
    "date": "12-08-2022'09:21"
    "message": "error message"
    "stacktrace": "stacktrace when in development mode"
  }
}
```

This controller advice give us a nice setup to add something like that easily in the future.

Reply:
And just remembered that with the `@ResponseStatus` annotations on exceptions above is no longer needed 😄